src/client/sw.js

```diff
@@ -3,13 +3,37 @@ let version = 'VERSION'
 workbox.setConfig({ debug: SW_DEBUG })
 workbox.skipWaiting()
 workbox.clientsClaim()
-workbox.core.setCacheNameDetails({ prefix: 'hn-pwa', suffix: `v2.${version}`, precache: 'precache', runtime: 'runtime' })
+
+// for precache we dont want to set versioning since every new version will start with empty cache
+// workbox already has a mechanism to update resources in precache based on their revision
+// this way only the resources that changed since last build will be replaced in cache
+workbox.core.setCacheNameDetails({ prefix: 'hn-pwa', precache: 'precache', runtime: `runtime-v2.${version}` })
```

Comment:
just for clarity - maybe worth noting with a comment at the top of the file and explaining that `VERSION` is set in webpack build so it's clearer how/when this will change